app/src/main/java/com/vinh/dictionary_1/utis/binding/BindingUtils.java

```diff
@@ -78,16 +80,30 @@ public static void toggleDrawerState(DrawerLayout drawerLayout, boolean isOpen)
 
     @BindingAdapter({ "fragment" })
     public static void setFragment(FrameLayout frameLayout, Fragment fragment) {
-        ((AppCompatActivity) frameLayout.getContext()).getSupportFragmentManager()
-                .beginTransaction()
-                .add(frameLayout.getId(), fragment, Constant.FRAGMENT_TAG_WORD_SEARCH_LIST)
+        FragmentManager fragmentManager =
+                ((AppCompatActivity) frameLayout.getContext()).getSupportFragmentManager();
+        if(fragment instanceof WordListFragment && fragmentManager.getFragments().size() > 1){
+            fragmentManager.beginTransaction()
+                    .show(fragment)
+                    .commit();
+            return;
+        }
+        fragmentManager.beginTransaction()
+                .add(frameLayout.getId(), fragment)
                 .addToBackStack(null)
                 .commit();
     }
 
     @BindingAdapter({ "textWatcher" })
     public static void setTextWatcher(EditText editText, TextWatcher textWatcher) {
-        editText.addTextChangedListener(textWatcher);
+        if (textWatcher != null) {
+            editText.addTextChangedListener(textWatcher);
+        }
+    }
+
+    @BindingAdapter({ "onTouchListener" })
+    public static void setOnTouchListener(EditText editText, View.OnTouchListener onTouchListener) {
+        editText.setOnTouchListener(onTouchListener);
```

Comment:
WARNING: [ClickableViewAccessibility] Custom view ``EditText`` has `setOnTouchListener` called on it but does not override `performClick`
 If a `View` that overrides `onTouchEvent` or uses an `OnTouchListener` does not also implement `performClick` and call it when clicks are detected, the `View` may not handle accessibility actions properly. Logic handling the click actions should ideally be placed in `View#performClick` as some accessibility services invoke `performClick` when a click action should occur.